x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/slm/SLMSnapshotBlockingIntegTests.java

```diff
@@ -0,0 +1,278 @@
+            client().admin().cluster().prepareReroute().get();
+            logger.info("--> waiting for snapshot to be deleted");
+            try {
+                SnapshotsStatusResponse s =
+                    client().admin().cluster().prepareSnapshotStatus(repoId).setSnapshots(completedSnapshotName).get();
+                assertNull("expected no snapshot but one was returned", s.getSnapshots().get(0));
+            } catch (SnapshotMissingException e) {
+                // Great, we wanted it to be deleted!
+            }
+        });
+
+        // Cancel/delete the snapshot
+        try {
+            client().admin().cluster().prepareDeleteSnapshot(repoId, secondSnapName).get();
+        } catch (SnapshotMissingException e) {
+            // ignore
+        }
+    }
+
+    private void initializeRepo(String repoName) {
+        client().admin().cluster().preparePutRepository(repoName)
+            .setType("mock")
+            .setSettings(Settings.builder()
+                .put("compress", randomBoolean())
+                .put("location", randomAlphaOfLength(6))
+                .build())
+            .get();
+    }
+
+    private void createSnapshotPolicy(String policyName, String snapshotNamePattern, String schedule, String repoId,
+                                      String indexPattern, boolean ignoreUnavailable) {
+        createSnapshotPolicy(policyName, snapshotNamePattern, schedule, repoId, indexPattern,
+            ignoreUnavailable, SnapshotRetentionConfiguration.EMPTY);
+    }
+
+    private void createSnapshotPolicy(String policyName, String snapshotNamePattern, String schedule, String repoId,
+                                      String indexPattern, boolean ignoreUnavailable,
+                                      SnapshotRetentionConfiguration retention) {
+        Map<String, Object> snapConfig = new HashMap<>();
+        snapConfig.put("indices", Collections.singletonList(indexPattern));
+        snapConfig.put("ignore_unavailable", ignoreUnavailable);
+        if (randomBoolean()) {
+            Map<String, Object> metadata = new HashMap<>();
+            int fieldCount = randomIntBetween(2,5);
+            for (int i = 0; i < fieldCount; i++) {
+                metadata.put(randomValueOtherThanMany(key -> "policy".equals(key) || metadata.containsKey(key),
+                    () -> randomAlphaOfLength(5)), randomAlphaOfLength(4));
+            }
+        }
+        SnapshotLifecyclePolicy policy = new SnapshotLifecyclePolicy(policyName, snapshotNamePattern, schedule,
+            repoId, snapConfig, retention);
+
+        PutSnapshotLifecycleAction.Request putLifecycle = new PutSnapshotLifecycleAction.Request(policyName, policy);
+        try {
+            client().execute(PutSnapshotLifecycleAction.INSTANCE, putLifecycle).get();
+        } catch (Exception e) {
+            fail("failed to create policy " + policy + " got: " + e);
```

Comment:
This doesn't capture the stack trace, which might be useful in the event of a failure. Can you add a `logger.error(e);` before the `fail()` call?